src/vm/methodtablebuilder.cpp

```diff
@@ -10432,7 +10432,8 @@ MethodTableBuilder::SetupMethodTable2(
     // Currently, only SIMD types have [Intrinsic] attribute
     //
     // We check this here fairly early to ensure other downstream checks on these types can be slightly more efficient.
-    if ((GetModule()->IsSystem() || GetAssembly()->IsSIMDVectorAssembly()) && IsValueClass() && bmtGenerics->HasInstantiation())
+    if ((GetModule()->IsSystem() || GetAssembly()->IsSIMDVectorAssembly()) && 
+        ((IsValueClass() && bmtGenerics->HasInstantiation()) || (IsAbstract() && IsSealed())))
```

Comment:
The `IsAbstract() && IsSealed()` condition does not look right. Could you please explain?

Reply:
I had to change the `[Intrinsic]` attribute to allow classes in order to put it on `public static class Vector`.  This is necessary because `Vector.IsHardwareAccelerated` is a specially recognized method (even more [specially recognized than the other intrinsics](https://github.com/dotnet/coreclr/blob/master/src/jit/simd.cpp#L765-L769)).

`IsAbstract() && IsSealed()` means it is a `static` class.  I didn't want to incur the attribute reading on all classes in `mscorlib`.  I can add this as a comment to the code.

Reply:
Well, we are checking the attribute for every method in CoreLib. So checking it on every type too would not be that big of a deal.

Reply:
Ideally, we would change the JIT to recognize the IsHardwareAccelerated method as intrinsic without needing to mark the static Vector type as instrinsic.

Reply:
Do you want me to remove the check for `IsValueClass()` as well?  Basically every type in mscorlib and System.Numerics.Vectors will be checked?

Or should I make the check something like:

```
    if ((GetModule()->IsSystem() || GetAssembly()->IsSIMDVectorAssembly()) &&
        ((IsValueClass() && bmtGenerics->HasInstantiation()) || (!IsEnum() && !IsInterface() && !IsDelegate())))
```

Which which should add checks for reference classes.

Reply:
> Basically every type in mscorlib and System.Numerics.Vectors will be checked?

I think it is fine. This condition is becoming impossible to reason about.